recipes-domu/domu-image-android/domu-image-android.bb

```diff
@@ -1,76 +1,127 @@
+SOC_REVISION = ""
 
-SRC_URI_append = " \
-    file://0001-fetch2-repo-Always-check-if-branch-is-correct.patch;patchdir=poky \
-    file://0002-fetch2-repo-Make-fetcher-always-sync-on-unpack.patch;patchdir=poky \
-    file://0003-fetch2-repo-Use-multiple-jobs-to-fetch-and-sync.patch;patchdir=poky \
-    file://0004-Make-pack-unpack-multithreaded.patch;patchdir=poky \
-    file://0005-fetch2-repo-Add-group-parameter.patch;patchdir=poky \
-"
 
-update_local_conf() {
-    local local_conf="${S}/build/conf/local.conf"
+# Product and variant are used to form string for 'lunch xenvm-userdebug'
+# These two are not suppoesed to be changed.
+ANDROID_PRODUCT = "xenvm"
+ANDROID_VARIANT = "userdebug"
 
-    cd ${S}
 
-    # override what xt-distro wants as machine as we will only use
-    # machines understood by Poky
-    base_update_conf_value ${local_conf} MACHINE "qemux86-64"
+# Disable all not used tasks
+# For android we use only
+# - do_fetch   -- get sources (repo init && sync)
+# - do_compile -- set android environment and build
+# - do_install -- copy assembled images into deploy folder
+# - do_build   -- default start point for build of domain
+do_unpack[noexec] = "1"
+do_patch[noexec] = "1"
+do_configure[noexec] = "1"
+do_populate_sdk[noexec] = "1"
+do_populate_sysroot[noexec] = "1"
+do_collect_build_history[noexec] = "1"
+do_populate_sstate_cache[noexec] = "1"
 
-    if [ ! -z ${XT_ANDROID_PREBUILDS_DIR} ];then
-        base_set_conf_value ${local_conf} DDK_KM_PREBUILT_MODULE "${XT_ANDROID_PREBUILDS_DIR}/pvr-km/pvrsrvkm.ko"
-        base_set_conf_value ${local_conf} DDK_UM_PREBUILDS "${XT_ANDROID_PREBUILDS_DIR}/pvr-um/"
-    fi
-}
+python __anonymous () {
+    if d.getVar("XT_ANDROID_PREBUILDS_DIR", expand=True):
+        DDK_KM_PREBUILT_MODULE = "${XT_ANDROID_PREBUILDS_DIR}/pvr-km/pvrsrvkm.ko"
+        DDK_UM_PREBUILDS = "${XT_ANDROID_PREBUILDS_DIR}/pvr-um/"
 
-python do_configure_append() {
-    bb.build.exec_func("update_local_conf", d)
+    prebuilt_vars = ["DDK_KM_PREBUILT_MODULE", "DDK_UM_PREBUILDS", "TARGET_PREBUILT_KERNEL"]
+    for var in prebuilt_vars:
+        if d.getVar(var, True) != None:
+            if not os.path.exists(d.getVar(var, True)):
+                raise bb.parse.SkipPackage('%s points to a non-existent path' % (var))
+            d.appendVar("PREBUILT_VARS", "%s=%s\n" % (var, d.getVar(var, True)))
 }
 
-# we do not have an SDK to populate
-do_populate_sdk() {
+# Note, that there is no need to set OUT_DIR_COMMON_BASE as
+# it is used as fallback option in case OUT_DIR is not set.
+OUT_DIR = "${SSTATE_DIR}/../${ANDROID_PRODUCT}-${SOC_FAMILY}"
+
+# This is important workaround.
+# Host python is used to build proper trusted application (TA) for OP-TEE.
+# We can't use android's python because we can't install required library into it.
+# So we have to provide host python with special library.
+# Also pay attention that PYTHON is set by poky, and not need to be set in host
+ANDROID_HOST_PYTHON="$(dirname ${PYTHON})"
```

Comment:
as i understand, the reason of host python usage is the libs: 
python3-pyelftools 
python3-pip
python3-crypto
pip3 pycryptodomex 
pip pycryptodome 

There are native-recipes to install them:
python3-pyelftools-native
python3-pycryptodome-native
python-pycryptodomex-native
python3-pycrypto-native

I think, it is reasonable try to use the native recipes, or it can not be implemented because of some reasons ?

Reply:
That's right, host python is used here to build trusted applications (TA) that need some specific crypto libraries.
So your proposal is correct. but it is still some kind of workaround.
Proper solution is to build TA outside of android build. In this case we can use mentioned native libraries and do not need to use host python.
From my point of view, separate build of TA is out of scope for this PR, and should be considered as separate task.